src/ApiPort/ConsoleProgressReporter.cs

```diff
@@ -77,6 +82,8 @@ public ConsoleProgressTask(string task, int? totalCount)
             {
                 await Task.Delay(1);
 
+#if RICH_CONSOLE
```

Comment:
I don't think there's any need to make a separate ConsoleProgressReporter since this is already #Ifdef'd out so easily. As it is now, everything works as before on Windows and on Ubuntu/Mac we just don't have the progress reporter's waiting animation (but the tool still seems perfectly usable without it).